src/Utilities/Compiler/Extensions/OperationBlockAnalysisContextExtension.cs

```diff
@@ -10,13 +12,41 @@ namespace Analyzer.Utilities.Extensions
 {
     internal static class OperationBlockAnalysisContextExtension
     {
-#pragma warning disable RS1012 // Start action has no registered actions.
         public static bool IsMethodNotImplementedOrSupported(this OperationBlockStartAnalysisContext context)
```

Comment:
This is already a complicated method and the underlying issue absolutely does not justify adding so much complexity to it.

Reply:
Shall I open a PR with the changes on the helper or do you think it's not worth it and we can redo them later if needs be?

Reply:
@Evangelink What changes are needed to helper? Is it just the removal of `RS1012` suppression?

Reply:
My question do we want to keep the helper to detect empty method/block? If so I will do a separate PR with it, if not then I don't do it (we don't have the need now).

Reply:
I would prefer to avoid any code changes to helper unless it is required by some caller.